File: leet-code/easy/cpp/0412 fizz-buzz.cpp
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ret(n);

        for(int i = 1; i <= n; i++) {
            if(i % 3 == 0) {
                ret[i - 1] += string("Fizz");
            }

            if(i % 5 == 0) {
                ret[i - 1] += string("Buzz");
            }

            if(ret[i - 1] == "") {
                ret[i - 1] += to_string(i);
            }
        }

        return ret;
    }
};

File: leet-code/go/0412 fizz-buzz.go
package leet

import "fmt"

func fizzBuzz(n int) []string {
	ans := make([]string, n)
	for i := 1; i <= n; i++ {
		a := ""

		if i%3 == 0 {
			a += "Fizz"
		}

		if i%5 == 0 {
			a += "Buzz"
		}

		if a == "" {
			a = fmt.Sprint(i)
		}

		ans[i-1] = a
	}

	return ans
}
